crates/key-value-redis/src/store.rs

```diff
@@ -55,90 +52,69 @@ impl StoreManager for KeyValueRedis {
 }
 
 struct RedisStore {
-    connection: Arc<Mutex<MultiplexedConnection>>,
+    connection: ConnectionManager,
     database_url: Url,
 }
 
 struct CompareAndSwap {
     key: String,
-    connection: Arc<Mutex<MultiplexedConnection>>,
+    connection: ConnectionManager,
     bucket_rep: u32,
 }
 
 #[async_trait]
 impl Store for RedisStore {
+    async fn after_open(&self) -> Result<(), Error> {
+        if let Err(_error) = self.connection.clone().ping::<()>().await {
+            //If IO error happens, ConnectionManager will start reconneciton in background
+            //so we do not take any action and just pray re-connection will be successful.
```

Comment:
Boy the enterprise adopters are gonna love that "just pray"

Is it worth emitting an info or trace log that the connection expired and we've pinged for a reconnect?

Reply:
I was going to and noticed that `key-value-redis` didn't log anywhere :trollface: 

Happy to bring log or tracing into this crate instead of gobbling the error.

Reply:
Nah no worries, leave it for now